test/addons/openssl-key-engine/binding.gyp

```diff
@@ -0,0 +1,41 @@
+{
+  'targets': [
+    {
+      'target_name': 'testkeyengine',
+      'type': 'none',
+      'includes': ['../common.gypi'],
+      'conditions': [
+        ['OS!="mac" and '
+         'OS!="axi" and '
+         'OS!="win" and '
+         'node_use_openssl=="true" and '
+         'node_shared=="false" and '
+         'node_shared_openssl=="false"', {
+          'type': 'shared_library',
+          'sources': [ 'testkeyengine.cc' ],
+          'product_extension': 'engine',
+          'include_dirs': ['../../../deps/openssl/openssl/include'],
+          'link_settings': {
+            'libraries': [
+              '../../../../out/<(obj_dir)/deps/openssl/<(openssl_product)'
```

Comment:
This isn't going to work on Travis or some of our Linux configurations (e.g. ARM) where compilation of the main `node` binary happens in a different job to running tests (i.e. the intermediate compiled artifacts for `deps` are not guaranteed to be available in the test jobs).